parser/tests/integration/test_parse_unparse_integration.py

```diff
@@ -21,17 +22,17 @@ def test_round_trip():
     session = test_parser.session
 
     # parse taxonomy
-    test_parser(TEST_TAXONOMY_TXT)
+    test_parser(TEST_TAXONOMY_TXT, "branch")
     # just quick check it runs ok with total number of nodes
-    query = "MATCH (n) RETURN COUNT(*)"
+    query = f"MATCH (n:{MULTI_LABEL}) RETURN COUNT(*)"
     result = session.run(query)
     number_of_nodes = result.value()[0]
     assert number_of_nodes == 13
     session.close()
 
     # dump taxonomy back
     test_dumper = unparser.WriteTaxonomy()
-    lines = list(test_dumper.iter_lines())
+    lines = list(test_dumper.iter_lines(MULTI_LABEL))
 
     original_lines = [line.rstrip("\n") for line in open(TEST_TAXONOMY_TXT)]
     # expected result is close to original file with a few tweaks
```

Comment:
With current test, we are not testing that there is no interference between to parsing.

It would be important to add a quick test where we parse twice the same taxonomy with two different branch name, and verify it's all ok on both taxonomy.

Reply:
@alexgarel I added a test (`test_two_branch_round_trip()`) for the same.